packages/api/src/platforms/vtex/resolvers/query.ts

```diff
@@ -33,9 +38,34 @@ export const Query = {
 
     const {
       loaders: { skuLoader },
+      clients: { commerce },
     } = ctx
 
-    return skuLoader.load(locator)
+    if (id) {
+      return skuLoader.load(id)
+    }
+
+    if (slug) {
+      try {
+        const skuId = slug?.split('-').pop() ?? ''
+        const sku = await skuLoader.load(skuId)
```

Comment:
I think we could be smarter here. Maybe trying to see if the `skuId` is a number and only if that's true we make the request. I think we could avoid lots of requests with this.

Reply:
the code will be ugly, but yeah